arb_os/signedTx.mini

```diff
@@ -15,50 +15,76 @@
 import func bytearray_setByte(ba: ByteArray, offset: uint, val: uint) -> ByteArray;
+import func bytearray_set256(ba: ByteArray, offset: uint, val: uint) -> ByteArray;
 import func bytearray_extract(ba: ByteArray, offset: uint, nbytes: uint) -> ByteArray;
 import func bytearray_unmarshalBytes(mb: MarshalledBytes) -> ByteArray;
 import func bytearray_marshalFull(ba: ByteArray) -> MarshalledBytes;
 import func marshalledBytes_hash(mb: MarshalledBytes) -> bytes32;
+import func bytearray_copy(
+    from: ByteArray,
+    fromOffset: uint,
+    to: ByteArray,
+    toOffset: uint,
+    nbytes: uint,
+) -> ByteArray;
 
+import func bytestream_new(ba: ByteArray) -> ByteStream;
+import func bytestream_skipBytes(ba: ByteStream, nbytes: uint) -> option<ByteStream>;
 
-import func rlp_encodeAndHashMessageInfo(
-    seqNum: uint,
-    gasPrice: uint,
-    gasLimit: uint,
-    to: address,
-    value: uint,
-    data: ByteArray,
-    v: uint,
-    r: uint,
-    s: uint
-) -> bytes32;
+import func rlp_encodeAndHashMessageInfo(tx: SignedTx) -> bytes32;
+import func rlp_decodeMessageInfo(inStream: ByteStream) -> option<(ByteStream, SignedTx)>;
 
 import impure func chainParams_chainId() -> uint;
 
 import func keccak256(ba: ByteArray, offset: uint, nbytes: uint) -> bytes32;
 
 
-type MessageFromL1 = struct {
+type IncomingRequest = struct {
     kind: uint,               // type of message
     blockNumber: uint,        // block number of the L1 block
     timestamp: uint,          // timestamp of the L1 block
     sender: address,          // address of the sender
-    inboxSeqNum: uint,        // sequence number in L1 inbox
+    inboxSeqNum: uint,        // sequence number in L1 inbox, or tx id
     msgData: MarshalledBytes  // contents of the message, as a marshalled bytearray
 }
 
-public impure func translateSignedTx(req: MessageFromL1) -> option<MessageFromL1> {
-    let extractedL2data = bytearray_unmarshalBytes(req.msgData);
+type SignedTx = struct {
+    seqNum: uint,
+    gasPrice: uint,
+    gasLimit: uint,
+    to: address,
+    value: uint,
+    data: ByteArray,
+    v: uint,
+    r: uint,
+    s: uint
+}
+
+public impure func translateSignedTx(req: IncomingRequest) -> option<IncomingRequest> {
+    // Given an IncomingRequest that contains an L2 signed tx message, translate the
+    //    L2 message from the RLP-encoded form it arrived in, into the old format which downstream code expects.
+    // Also, check the signature on the tx, and set:
+    //      - the sender field in the request equal to signer, and
+    //      - the requestID field in the request equal to the Ethereum-compatible tx id,
+    // Return the result, or None if message is malformed or the signature is invalid.
+
+    let rlpData = bytearray_unmarshalBytes(req.msgData);
+    rlpData = bytearray_extract(rlpData, 1, bytearray_size(rlpData)-1);
+    let (_, signedTx) = rlp_decodeMessageInfo(bytestream_new(rlpData))?;
+
+    let extractedL2data = signedTxToBytes(signedTx);
```

Comment:
I know you might refactor this further, but one easy thing to do would be to change this to `signedTxToUnsignedTxBytes` and then we wouldn't need the following code

Reply:
Done.